frontend/file/src/main/java/com/virtuslab/gitmachete/frontend/file/quickfix/CreateBranchQuickFix.java

```diff
@@ -0,0 +1,72 @@
+package com.virtuslab.gitmachete.frontend.file.quickfix;
+
+import static com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle.getNonHtmlString;
+
+import java.util.Collections;
+
+import com.intellij.codeInsight.intention.IntentionAction;
+import com.intellij.codeInspection.util.IntentionFamilyName;
+import com.intellij.codeInspection.util.IntentionName;
+import com.intellij.openapi.editor.Editor;
+import com.intellij.openapi.project.Project;
+import com.intellij.psi.PsiFile;
+import com.intellij.util.IncorrectOperationException;
+import git4idea.branch.GitBrancher;
+import git4idea.repo.GitRepository;
+import io.vavr.control.Option;
+import lombok.experimental.ExtensionMethod;
+
+import com.virtuslab.gitmachete.frontend.file.MacheteFileUtils;
+import com.virtuslab.gitmachete.frontend.resourcebundles.GitMacheteBundle;
+import com.virtuslab.qual.guieffect.UIThreadUnsafe;
+
+@ExtensionMethod({GitMacheteBundle.class})
+public class CreateBranchQuickFix implements IntentionAction {
+
+  private final String branch;
+  private final String parentBranch;
+  private final PsiFile macheteFile;
+
+  public CreateBranchQuickFix(String processedBranchName, String parentBranchName, PsiFile processedFile) {
+    branch = processedBranchName;
+    parentBranch = parentBranchName;
+    macheteFile = processedFile;
+  }
+
+  @Override
+  public @IntentionName String getText() {
+    return getNonHtmlString("action.GitMachete.BaseSlideInBelowAction.dialog.create-new-branch.title")
+        .format(parentBranch) + ": " + branch;
+  }
+
+  @Override
+  public @IntentionFamilyName String getFamilyName() {
+    return "Git Machete";
+  }
+
+  @Override
+  public boolean isAvailable(Project project, Editor editor, PsiFile file) {
+    return true;
+  }
+
+  @Override
+  @UIThreadUnsafe
+  public void invoke(Project project, Editor editor, PsiFile file) throws IncorrectOperationException {
+    createNewBranchFromParent(project);
+  }
+
+  @Override
+  public boolean startInWriteAction() {
+    return false;
+  }
+
+  @UIThreadUnsafe
+  private void createNewBranchFromParent(Project project) {
+    Option<GitRepository> gitRepositoryOption = MacheteFileUtils.findGitRepositoryForPsiMacheteFile(macheteFile);
+    if (gitRepositoryOption.isDefined()) {
+      GitBrancher.getInstance(project).createBranch(branch, Collections.singletonMap(gitRepositoryOption.get(), parentBranch));
```

Comment:
Hmmm would it be possible to trigger a reparsing (?) of the machete file once the branch is created? There seems to be a delay after creating the branch before the red squiggle disappears. Ofc skip if complicated, not that important

Reply:
I tried to add a solution to trigger a reparsing but failed for now. Maybe it would be possible, but:
- either the solution did not work at all
- or solution did work, but caused multiple recursive calls of `MacheteAnnotator#annotate` even if nothing changed
I think I will add an issue to fix that. It is probably posssible, but somehow tricky and needs some work.

Reply:
Okay, pls open an issue then

Reply:
issue created: https://github.com/VirtusLab/git-machete-intellij-plugin/issues/1281